runtime/src/commonMain/kotlin/pbandk/internal/json/JsonValueMarshaller.kt

```diff
@@ -105,4 +106,15 @@ internal class JsonValueMarshaller(private val jsonConfig: JsonConfig) {
             writeValue(k, keyType).content to writeValue(v, valueType)
         }
     }
+
+    fun writeValue(value: Value): JsonElement = when (value.kind) {
+        is Value.Kind.StringValue -> writeString(value.kind.value)
+        is Value.Kind.BoolValue -> writeBool(value.kind.value)
+        is Value.Kind.NumberValue -> writeDouble(value.kind.value)
+        is Value.Kind.StructValue -> writeValue(value.kind.value, Struct.fieldDescriptors[0].type)
+        is Value.Kind.ListValue -> jsonArray {
+            for (elem in value.kind.value.values) +writeValue(elem)
+        }
+        is Value.Kind.NullValue, null -> JsonNull
```

Comment:
🔧 It's possible for the `NullValue` enum to be used outside the context of `Value` and it should still be properly serialized as `null` in those cases. To make that possible, you should modify the `writeValue()` method above to something like:

```kt
fun writeValue(value: Any, type: FieldDescriptor.Type): JsonElement = when (type) {
    ...
    is FieldDescriptor.Type.Enum<*> -> when (type.enumCompanion) {
        NullValue -> JsonNull
        else -> writeEnum(value as Message.Enum)
    }
    ...
```